### DISCUSSION TITLE
more options to query device data?

### DISCUSSION THREAD

**Discussion by LFrank2021** [opendcim/openDCIM#1505] [Q&A] [answered]
Hi,

is there any way to get sensor data other than with SNMP?
We have some new devices (UPS) that could be queried using REST-API (JSON).
I am doing that for general monitoring purposes but would like to present that data within openDCIM, to get the full overview (measured power).

Kind regards
Frank

**Comment by wilpig** [comment]
No.  There aren't methods for updating that information outside of the process that is tied to SNMP.

https://github.com/opendcim/openDCIM/blob/master/classes/PowerDistribution.class.php#L381

**Comment by samilliken** [comment, marked as answer]
To clarify upon @wilpig's comment - we don't provide any other methods.   However, there is an API that will allow you to update the sensor settings from your own script.   There are PUT methods for both /pdustats and /sensorreadings

**Comment by samilliken** [comment]
Correction, POST method.   Anyway, it's documented in Swagger.

**Comment by LFrank2021** [comment]
Thanks, that could become helpful.
